authlib/integrations/base_client/sync_app.py

```diff
@@ -291,7 +291,12 @@ def request(self, method, url, token=None, **kwargs):
     def load_server_metadata(self):
         if self._server_metadata_url and '_loaded_at' not in self.server_metadata:
             with self.client_cls(**self.client_kwargs) as session:
-                resp = session.request('GET', self._server_metadata_url, withhold_token=True)
+                resp = session.request(
+                    'GET',
+                    self._server_metadata_url,
+                    withhold_token=True,
+                    **self.client_kwargs
```

Comment:
❓ Do we feel confident that:
* `client_kwargs` is a good place for setting request arguments?
* and all fields which are set in `client_kwargs` should be passed to request?

Reply:
I am not 100% confident. Do you have a better idea?

Might be better to just bring up with the maintainers to be honest, before investing too much time going down different paths.

Reply:
Yeah I agree that we should get their take on it before we invest too much. Other options I would consider instead of relying directly on what's passed in `client_kwargs` though:
* Add `request_kwargs` to `client_kwargs`:
```python3
client_kwargs: {
    'request_kwargs': {
        'timeout': timeout,
        ...,
    }
}
```
* Or add it to `oauth.register()`:
```python3
oauth = OAuth()
oauth.register(
    name='google',
    server_metadata_url=GOOGLE_OIDC_METADATA_URL,
    authorize_url=GOOGLE_AUTHORIZE_URL,
    client_kwargs=GOOGLE_EXTRA_CLIENT_ARGUMENTS,
    request_kwargs=REQUEST_KWARGS,
)
```
Which I think would deal with that this solution seems a bit implicit. 

But maybe passing all `client_kwargs` to the request makes sense? To me the meaning of `client_kwargs` is a bit fuzzy, I haven't spent time digging into it.